tools/port-drayage-webservice/src/main/java/com/leidos/inspection/InspectionActions.java

```diff
@@ -0,0 +1,194 @@
+     *         id
+     */
+    public InspectionStatus getInspectionStatus(String vehicleId) {
+        if (vehicleId != null) {
+            // Pending actions ( null check since arraylist is initially null )
+            if (pendingInspection.getInspections() != null) {
+                for (InspectionStatus inspectionStatus : pendingInspection.getInspections()) {
+                    if (vehicleId.equals(inspectionStatus.getVehicleId())) {
+                        return inspectionStatus;
+                    }
+                }
+            }
+            // Current action
+            if (currentInspection != null && vehicleId.equals(currentInspection.getVehicleId())) {
+                return currentInspection;
+            }
+            // Completed actions ( null check since arraylist is initially null )
+            if (completedInspections.getInspections() != null) {
+                for (InspectionStatus inspectionStatus : completedInspections.getInspections()) {
+                    if (vehicleId.equals(inspectionStatus.getVehicleId())) {
+                        return inspectionStatus;
+                    }
+                }
+            }
+            logger.warn(String.format("No current inspection action for vehicle ID %s !", vehicleId));
+            return null;
+        }
+        logger.warn("Null vehicle id is not valid!");
+        return null;
+    }
+
+    /**
+     * To complete inspection set {@link InspectionActions#currentInspection} status
+     * to {@link StatusEnum#PASSED}. Current inspection is then added to the
+     * completed inspections list and the first pending action , if any are present
+     * is set as the new current action. If no pending actions are present the
+     * current action is set to null.
+     * 
+     */
+    public void completeInspection() {
+        if (currentInspection != null) {
+            // Set current inspection to PASSED
+            currentInspection.setStatus(StatusEnum.PASSED);
+            currentInspection.setCompleted(System.currentTimeMillis());
+            completedInspections.addInspectionsItem(currentInspection);
+            // If there are any pending actions set them to current
+            if (pendingInspection.getInspections() != null && !pendingInspection.getInspections().isEmpty()) {
+                currentInspection = pendingInspection.getInspections().remove(0);
+            }
+            // else set current to null
+            else {
+                currentInspection = null;
+            }
+
+        } else
+            logger.warn("No current inspection in progress!");
+    }
+
+    /**
+     * To indicate that the vehicle under inspection is required to proceed to the
+     * holding area for further inspection an operator will use this request to set
+     * {@link InspectionActions#currentInspection} status to.
+     * {@link StatusEnum#PROCEED_TO_HOLDING}.
+     */
+    public void proceedToHolding() {
```

Comment:
So this may be more clear using the action diagram I created for the service but essentially this is for the UI form submission to post to. It triggers the state of the inspection to be proceed to holding, which will trigger V2X-Hub to provide the vehicle with an action to drive to the holding area. Once at the holding area the vehicle will send confirmation that it has arrived and V2X-Hub will sent a POST request to change its status to HOLDING indicating it is at the holding area as oppose to driving to it. At that point the UI will allow the inspection personnel to pass it, since passing it while its holding would not really make sense.